xwiki-rendering-macros/xwiki-rendering-macro-message/src/main/java/org/xwiki/rendering/internal/macro/message/AbstractMessageMacro.java

```diff
@@ -62,23 +54,29 @@
     public AbstractMessageMacro(String macroName, String macroDescription)
     {
         super(macroName, macroDescription,
-            new DefaultContentDescriptor("Content of the message", true, Block.LIST_BLOCK_TYPE));
+            new DefaultContentDescriptor("Content of the message", true, Block.LIST_BLOCK_TYPE),
+            BoxMacroParameters.class);
     }
 
     @Override
-    public List<Block> execute(Object parameters, String content, MacroTransformationContext context)
-        throws MacroExecutionException
+    public List<Block> execute(BoxMacroParameters parameters, String content,
+        MacroTransformationContext context) throws MacroExecutionException
     {
         if (StringUtils.isEmpty(content)) {
             throw new MacroExecutionException(CONTENT_MISSING_ERROR);
         }
 
-        BoxMacroParameters boxParameters = new BoxMacroParameters();
-
-        boxParameters.setCssClass(context.getCurrentMacroBlock().getId() + "message");
+        parameters.setCssClass(context.getCurrentMacroBlock().getId() + "message");
```

Comment:
I'm not a big fan of modifying the input parameters object.

A cleaner solution might be to overwrite getClassProperty() and return something like `super.getClassProperty() + ' ' + context.getCurrentMacroBlock().getId() + "message"`. This also allow message macro used to indicate a custom class which is a nice news feature.

Reply:
Not at all. I did not suggested to introduce anything new, just to not break BoxMacroParameters properties.

Reply:
Not sure to follow. Here the class is dedicated to message macro, as the name suggest, so it feels wrong to put it as default for box macro...

Reply:
My bad I mixed up between getClassProperty and getCssClass. OK I'll do that.